library/core/src/any.rs

```diff
@@ -81,11 +85,77 @@
 //!     do_work(&my_i8);
 //! }
 //! ```
+//!
+//! # `Provider` and `Demand`
+//!
+//! `Provider` and the associated APIs support generic, type-driven access to data, and a mechanism
+//! for implementers to provide such data. The key parts of the interface are the `Provider`
+//! trait for objects which can provide data, and the [`request_value`] and [`request_ref`]
+//! functions for requesting data from an object which implements `Provider`. Generally, end users
+//! should not call `request_*` directly, they are helper functions for intermediate implementers
+//! to use to implement a user-facing interface.
```

Comment:
pondering: would it be worth elaborating on *why* they shouldn't be called directly?  It sure looks like they *could* be called directly no problem?